contracts/modules/common/TokenPriceProvider.sol

```diff
@@ -16,33 +16,41 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity ^0.6.10;
 import "@openzeppelin/contracts/math/SafeMath.sol";
-import "../../../lib/other/ERC20.sol";
+import "../../../lib/other/IERC20Extended.sol";
 import "../../infrastructure/base/Managed.sol";
 
 contract TokenPriceProvider is Managed {
     using SafeMath for uint256;
 
     mapping(address => uint256) public cachedPrices;
 
-    function setPrice(ERC20 _token, uint256 _price) public onlyManager {
-        cachedPrices[address(_token)] = _price;
-    }
-
-    function setPriceForTokenList(ERC20[] calldata _tokens, uint256[] calldata _prices) external onlyManager {
-        for (uint16 i = 0; i < _tokens.length; i++) {
-            setPrice(_tokens[i], _prices[i]);
-        }
-    }
-
     /**
      * @dev Converts the value of _amount tokens in ether.
      * @param _amount the amount of tokens to convert (in 'token wei' twei)
-     * @param _token the ERC20 token contract
+     * @param _token the token address
      * @return the ether value (in wei) of _amount tokens with contract _token
      */
     function getEtherValue(uint256 _amount, address _token) external view returns (uint256) {
-        uint256 decimals = ERC20(_token).decimals();
+        uint8 decimals;
+
+        try IERC20Extended(_token).decimals() returns (uint8 _decimals) {
+            decimals = _decimals;
+        }
+        catch (bytes memory /*lowLevelData*/) {
+        }
+
         uint256 price = cachedPrices[_token];
-        return price.mul(_amount).div(10**decimals);
+        uint256 etherValue = decimals == 0 ? price.mul(_amount) : price.mul(_amount).div(10**uint(decimals));
```

Comment:
Maybe drop the condition? 
```
uint256 etherValue = price.mul(_amount).div(10**uint(decimals));
```
should work for all types of tokens.

Reply:
Probably just saving some gas but maybe clarity is worth more.